ldaptive/src/main/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDao.java

```diff
@@ -337,22 +337,28 @@ public void removeFromGroup(User user, Group group) {
     }
 
     protected void modifyDn(final Connection conn, final ModifyDnRequest request) throws LdapException {
-        try {
-            new ModifyDnOperation(conn).execute(request);
-        } catch (final LdapException e) {
-            switch (e.getResultCode()) {
-                case OTHER:
-                    // this is a partition is busy error, let's pause, then retry
-                    try {
-                        Thread.sleep(100L);
-                    } catch (final InterruptedException ignored) {
+        boolean complete = false;
+        while (!complete) {
+            try {
+                new ModifyDnOperation(conn).execute(request);
+                complete = true;
+            } catch (final LdapException e) {
+                if (e.getResultCode() != null) {
```

Comment:
Since it's conceivable that null result code could mean partition busy, should we bother about that? (since of course null result code could be some other exception).